lib/fastlane/plugin/wpmreleasetoolkit/actions/common/create_release_backmerge_pull_request_action.rb

```diff
@@ -83,6 +83,14 @@ def self.determine_target_branches(source_release_version:, default_branch:)
       # @param team_reviewers [Array<String>] the team reviewers for the pull request.
       # @return [void]
       def self.create_backmerge_pr(token:, repository:, title:, head_branch:, base_branch:, labels:, milestone:, reviewers:, team_reviewers:)
+        # Check for differences between head_branch and base_branch
+        diff_output = Actions.sh('git', 'diff', "#{base_branch}..#{head_branch}").strip
```

Comment:
I'm not 100% sure this will still work for cases where the `base_branch` has evolved (i.e. got new commits) since the head_branch got cut and the PR got created 🤔  We might need to first find the `git merge-base #{base_branch} #{head_branch}` first, [or alternatively use the three-dots syntax instead of two-dots range](https://stackoverflow.com/questions/7251477/what-are-the-differences-between-double-dot-and-triple-dot-in-git-dif).
Though it's late and I'm not completely sure which one of the two we actually need (and which one of the two cases GitHub will consider it to create an empty PR). I _think_ the diff that GitHub shows on PRs is the `...` one not the `..` one, and thus we should probably check the `...` one for emptiness… but will probably be worth experimenting a bit to confirm and double-check (and document via code comments)

Also, since we're only interested if there's an actual diff or not, we don't really need the diff's actual output. So we should instead use `git diff --quiet` and check the exit code, rather than using `git diff` and checking if the output is empty.

Reply:
Also, I wonder if we shouldn't start using the `Git` gem, instead of shelling out to `sh('git', …)`… though that is probably a broader discussion for outside this PR, because I know we already use `sh('git', …)` in our git helpers so far too… so we'd probably want to migrate everything to using the `Git` gem to be consistent if we do that.

Reply:
Nice catch! I just seen [this](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-comparing-branches-in-pull-requests#three-dot-and-two-dot-git-diff-comparisons), which suggests to indeed use three-dot diff instead:
> By default, pull requests on GitHub show a three-dot diff.

---

On Git operations:

> So we'd probably want to migrate everything to using the Git gem to be consistent if we do that

That's an interesting point -- I think it would be nice to be consistent. Using the `Git` gem would be nice, but I also see it as a more complex project as I guess we'd have to implement more Ruby code to do things like filtering, besides the changes we'd also have to change quite some tests, etc. I've created https://github.com/wordpress-mobile/release-toolkit/issues/572 so we can have a starting point for discussion.

Reply:
> So we should instead use git diff --quiet and check the exit code

The problem with that is that `sh()` fails the lane when there's a 1 exit code.

Reply:
Also, I had plans to add a helper in release-toolkit that would return of there has been new commits between two points in git. The typical use case would be to check if there had been new commits on `release/*` since last beta git tag, so we can determine with automation if a new beta is necessary or not. We could then allow us to run `new_beta_release` on a daily scheduled build on CI, exciting early and not doing the beta if no new commits were detected.

(I have a roadmap item for this)

--/

This PR could actually be the occasion to start implementing that helper method here and use it to detect if the PR will be empty rather than looking at the diff?

Also because now that I'm thinking about it, I wonder if GitHub wouldn't allow a PR with an empty diff as long as the PR would create a non-empty merge commit. A case where this can happen is if you create a commit on your head branch then push another commit reverting the change of the first. The diff would be empty but the git graph after the merge would still show the 2 commits being merged into the target branch in the end so that would still be a valid merge (as opposed to trying to merge A into B when A and B end up pointing to the same commit) 🤔 

Definitively worth some experimentation with those edge cases to see what works.

Reply:
> (I have a roadmap item for this)

Found it: https://github.com/orgs/Automattic/projects/841/views/1?filterQuery=Beta&pane=issue&itemId=51038655

The idea I had back then for such detection was to use `git rev-list --count --first-parent ${previous_tag}..HEAD` to count the number of commits between `previous_tag` and `HEAD` (and consider the diff empty if that count is 0).
Haven't tested that idea in practice though.

Reply:
> The problem with that is that `sh()` fails the lane when there's a 1 exit code.

Right. IINM we can [pass a block to `sh`](https://github.com/fastlane/fastlane/blob/2c34057215aeabdaac4aea7806a8f166417c8dff/fastlane/lib/fastlane/helper/sh_helper.rb#L22) to catch those error codes and prevent fastlane to `UI.user_error!` on those occasions, though it would still `UI.error` (i.e. print the message in red in the logs, but at least not fail the lane and continue instead).

Reply:
WFM 👍

Reply:
I updated the code to follow this approach on 8e5aa9e807543fa76bfb705ca080136376d73e8e + 7aa838b7138bff644d7458aa3b846775e7e58fec.